asapdiscovery-modeling/asapdiscovery/modeling/scripts/prep_fragalysis_for_plipify.py

```diff
@@ -0,0 +1,1131 @@
+        List of parsed crystal structures
+    """
+    from pathlib import Path
+
+    import pandas
+    from tqdm import tqdm
+
+    x_dir = Path(x_dir)
+
+    df = pandas.read_csv(x_fn)
+
+    # Only keep rows of dataframe where the name_filter_column includes the name_filter string
+    if name_filter:
+        if type(name_filter) == str:
+            idx = df[name_filter_column].apply(lambda x: name_filter in x)
+            df = df[idx]
+        elif type(name_filter) == list:
+            for filter in name_filter:
+                idx = df[name_filter_column].apply(lambda x: filter in x)
+                df = df[idx]
+    # Drop duplicates, keeping only the first one.
+    if drop_duplicate_datasets:
+        df = df.drop_duplicates("RealCrystalName")
+
+    # Build argument dicts for the CrystalCompoundData objects
+    xtal_dicts = [
+        dict(zip(("smiles", "dataset", "compound_id"), r[1].values))
+        for r in df.loc[:, ["smiles", "crystal_name", "alternate_name"]].iterrows()
+    ]
+
+    # Add structure filename information and filter if not found
+    filtered_xtal_dicts = []
+    for d in tqdm(xtal_dicts):
+        glob_str = f"{d['dataset']}*/*.pdb"
+        fns = list(x_dir.glob(glob_str))
+        for fn in fns:
+            d["str_fn"] = str(fn)
+
+            # This should basically always be true since we're getting the filenames from glob but just in case.
+            if os.path.isfile(fn):
+                filtered_xtal_dicts.append(d)
+    assert (
+        len(filtered_xtal_dicts) > 0
+    ), "No structure filenames were found by parse_fragalysis"
+
+    # Build CrystalCompoundData objects for each row
+    print(f"Loading {len(filtered_xtal_dicts)} structures")
+    xtal_compounds = [CrystalCompoundData(**d) for d in filtered_xtal_dicts]
+
+    return xtal_compounds
+
+
+########################################################################################################################
+########################################################################################################################
+
+
+def check_completed(d, prefix):
+    """
+    Check if this prep process has already been run successfully in the given
+    directory.
+
+    Parameters
+    ----------
+    d : str
+        Directory to check.
+
+    Returns
+    -------
+    bool
+        True if both files exist and can be loaded, otherwise False.
+    """
+
+    if (not os.path.isfile(os.path.join(d, f"{prefix}_prepped_receptor_0.oedu"))) or (
+        not os.path.isfile(os.path.join(d, f"{prefix}_prepped_receptor_0.pdb"))
+    ):
+        return False
+
+    try:
+        du = oechem.OEDesignUnit()
+        oechem.OEReadDesignUnit(
+            os.path.join(d, f"{prefix}_prepped_receptor_0.oedu"), du
+        )
+    except Exception:
+        return False
+
+    try:
+        _ = load_openeye_pdb(os.path.join(d, f"{prefix}_prepped_receptor_0.pdb"))
+    except Exception:
+        return False
```

Comment:
We have a couple of generic exception being caught here, while it might be hard to know what exact exception will/can be raised from OpenEye, we want to at least output the error/exception message such as:

```python
except Exception as e:
    warning(str(e))
    ...
```

Reply:
related to #185 but we don't need to fix this in this file as it is just there in case.